app/javascript/components/KeywordSearch.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react';
+import Button from 'react-bootstrap/lib/Button';
+import InputGroup from 'react-bootstrap/lib/InputGroup';
+import Form from 'react-bootstrap/lib/Form';
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+
+class KeyWordSearch extends React.Component{
```

Comment:
We should probably migrate to function components and manage state with [hooks](https://reactjs.org/docs/hooks-overview.html).  They're more modern and considerably more succinct.  Check out [FiltersBox.js](https://github.com/broadinstitute/single_cell_portal_core/blob/2857ce9af09d961e25ac30048446d0b42a6dee79/app/javascript/components/FiltersBox.js#L13) for an example.

There's an open question on when we migrate, so I don't consider this style blocking.

Reply:
Your examples have been a great help! I will be migrating to function components for [SCP-2321](https://broadworkbench.atlassian.net/browse/SCP-2123) and [SCP-2124](https://broadworkbench.atlassian.net/browse/SCP-2124).